src/Doctrine/Orm/Filter/AttributeFilter.php

```diff
@@ -0,0 +1,272 @@
+        foreach (array_keys($properties) as $property) {
+            $description += $this->getFilterDescription($property, self::ATTRIBUTE_ID);
+            $description += $this->getFilterDescription($property, self::VALUE);
+        }
+
+        return $description;
+    }
+
+    protected function getFilterDescription(string $fieldName, string $operator): array
+    {
+        /** @var string $propertyName */
+        $propertyName = $this->normalizePropertyName($fieldName);
+
+        return [
+            sprintf('%s[%s]', $propertyName, $operator) => [
+                'property' => $propertyName,
+                'type' => 'string',
+                'required' => false,
+            ],
+        ];
+    }
+
+    private function getAttributeId(array $values, string $property): ?string
+    {
+        if (array_key_exists(self::ATTRIBUTE_ID, $values)) {
+            /** @var string $attributeId */
+            $attributeId = $values[self::ATTRIBUTE_ID];
+        } else {
+            $this->getLogger()->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(
+                    sprintf(
+                        '%s is required for "%s" property',
+                        self::ATTRIBUTE_ID,
+                        $property
+                    )
+                ),
+            ]);
+
+            return null;
+        }
+
+        return $attributeId;
+    }
+
+    private function extractValue(array $values, string $property): ?string
+    {
+        if (array_key_exists(self::VALUE, $values)) {
+            /** @var string $value */
+            $value = $values[self::VALUE];
+        } else {
+            $this->getLogger()->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(
+                    sprintf(
+                        '%s is required for "%s" property',
+                        self::VALUE,
+                        $property
+                    )
+                ),
+            ]);
+
+            return null;
+        }
+
+        return $value;
+    }
+
+    /**
+     * @param ?mixed $value
+     *
+     * @return int|float|string|bool|\DateTime|null
+     *
+     * @throws \Exception
+     */
+    private function normalizeValue($value, string $type)
+    {
+        switch ($type) {
+            case CheckboxAttributeType::TYPE:
+                $value = (bool) $value;
+
+                break;
+            case DateAttributeType::TYPE:
+            case DatetimeAttributeType::TYPE:
+                Assert::string($value);
+                $value = new \DateTime($value);
+
+                break;
+            case IntegerAttributeType::TYPE:
+                $value = (int) $value;
+
+                break;
+            case PercentAttributeType::TYPE:
+                $value = (float) $value;
+
+                break;
+            case SelectAttributeType::TYPE:
+                //assume IRI ?
+                //TODO::
```

Comment:
What about this?

Reply:
At first, I thought that might be select across other entities, but after checking, it will be only enum strings